esmvaltool/diag_scripts/ocean/diagnostic_tools.py

```diff
@@ -211,13 +349,25 @@ def add_legend_outside_right(
 
     # Add emply plots to dummy axis.
     for index in sorted(plot_details.keys()):
-        colour = plot_details[index]['c']
+        try:
```

Comment:
I believe we discussed this code in a previous PR and now you're changing it back again..

Reply:
Please see https://github.com/ESMValGroup/ESMValTool/pull/711#pullrequestreview-175426941 and https://github.com/ESMValGroup/ESMValTool/pull/711#pullrequestreview-175751154

Reply:
Ah yes, I don't know how those changes got lost. I've brought them back in now. Sorry to make the extra work for you @bouweandela.